src/js/state/Current/selectors.ts

```diff
@@ -34,34 +35,40 @@ export const getLocation = (state: State) => {
 
 export const getQueryLocationData = (
   state: State
-): {queryId: string | null} => {
-  type Params = {queryId?: string}
+): {queryId: string | null; version: string | null} => {
+  type Params = {queryId?: string; version?: string}
   const match = matchPath<Params>(getLocation(state).pathname, [
-    "/lakes/:lakeId/queries/:queryId",
+    queryVersion.path,
+    query.path,
   ])
   const queryId = match?.params?.queryId
-  return {queryId}
+  const version = match?.params?.version
+  return {queryId, version}
 }
 
-const getRawQuery = (state: State): Query | null => {
-  const {queryId} = getQueryLocationData(state)
-  if (!queryId) return null
-  // query lookup policy is to search drafts first, then local, and finally remote
-  const query =
-    DraftQueries.getById(queryId)(state) ||
-    Queries.getQueryById(queryId)(state) ||
-    RemoteQueries.getQueryById(queryId)(state)
-  if (!query) return null
-  return query
-}
-
-export const getQuery = createSelector<State, Query | null, BrimQuery>(
-  getRawQuery,
-  (query) => {
+export const getQueryById =
```

Comment:
this is a generic query selector (i.e. it will select from either of the 3 query type slices (draft, local, remote), so I wasn't sure which selector code to put it and just settled on Current for now. Open to other suggestions.